contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,337 @@
+    kyberProxy = _kyberProxy;
+  }
+
+  function changeRewardsDistributor(IRewardsDistributor _rewardsDistributor) external onlyAdmin {
+    rewardsDistributor = _rewardsDistributor;
+  }
+
+  function changeGovernance(IKyberGovernance _kyberGovernance) external onlyAdmin {
+    kyberGovernance = _kyberGovernance;
+  }
+
+  function changeFees(
+    uint256 _mintFeeBps,
+    uint256 _claimFeeBps,
+    uint256 _burnFeeBps
+  ) external onlyAdmin {
+    _changeFees(_mintFeeBps, _claimFeeBps, _burnFeeBps);
+  }
+
+  function depositWithOldKnc(uint256 tokenWei) external {
+    oldKnc.safeTransferFrom(msg.sender, address(this), tokenWei);
+    INewKNC(address(newKnc)).mintWithOldKnc(tokenWei);
+    _deposit(tokenWei, msg.sender);
+  }
+
+  function depositWithNewKnc(uint256 tokenWei) external {
+    newKnc.safeTransferFrom(msg.sender, address(this), tokenWei);
+    _deposit(tokenWei, msg.sender);
+  }
+
+  /*
+   * @notice Called by users burning their token
+   * @dev Calculates pro rata KNC and redeems from staking contract
+   * @param tokensToRedeem
+   */
+  function withdraw(uint256 tokensToRedeemTwei) external nonReentrant {
+    require(balanceOf(msg.sender) >= tokensToRedeemTwei, 'insufficient balance');
+
+    uint256 proRataKnc = getLatestStake().mul(tokensToRedeemTwei).div(totalSupply());
+    _unstake(proRataKnc);
+    proRataKnc = _administerAdminFee(FeeTypes.BURN, proRataKnc);
+    super._burn(msg.sender, tokensToRedeemTwei);
+
+    newKnc.safeTransfer(msg.sender, proRataKnc);
+  }
+
+  /*
+   * @notice Vote on KyberDAO campaigns
+   * @dev Admin calls with relevant params for each campaign in an epoch
+   * @param proposalIds: DAO proposalIds
+   * @param optionBitMasks: corresponding voting options
+   */
+  function vote(uint256[] calldata proposalIds, uint256[] calldata optionBitMasks)
+    external
+    onlyOperator
+  {
+    require(proposalIds.length == optionBitMasks.length, 'invalid length');
+    for (uint256 i = 0; i < proposalIds.length; i++) {
+      kyberGovernance.submitVote(proposalIds[i], optionBitMasks[i]);
+    }
+  }
+
+  /*
+   * @notice Claim accumulated reward thus far
+   * @notice Will apply admin fee to KNC token.
+   * Admin fee for other tokens applied after liquidation to KNC
+   * @dev Admin or operator calls with relevant params
+   * @param cycle - sourced from Kyber API
+   * @param index - sourced from Kyber API
+   * @param tokens - ERC20 fee tokens
+   * @param merkleProof - sourced from Kyber API
+   */
+  function claimReward(
+    uint256 cycle,
+    uint256 index,
+    IERC20Ext[] calldata tokens,
+    uint256[] calldata cumulativeAmounts,
+    bytes32[] calldata merkleProof
+  ) external onlyOperator {
+    rewardsDistributor.claim(cycle, index, address(this), tokens, cumulativeAmounts, merkleProof);
+
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == newKnc) {
+        uint256 availableKnc = _administerAdminFee(
```

Comment:
why not just check availableKnc > 0 then stake?

Reply:
as discussed over TG, fixed